src/coreEnforcer.ts

```diff
@@ -610,6 +619,10 @@ export class CoreEnforcer {
    * @return whether to allow the request and the reason rule.
    */
   public async enforceEx(...rvals: any[]): Promise<[boolean, string[]]> {
-    return generatorRunAsync(this.privateEnforce(true, true, ...rvals));
+    return generatorRunAsync(this.privateEnforce(true, true, this.enforceContext, ...rvals));
+  }
+
+  public useEnforceContext(rType: string, pType: string, eType: string, mType: string): void {
```

Comment:
Wrong implementation, please remove this method, use the first parameter in `enforce` as the context, you should check the `ravals[0]`, if `ravals[0]` is context type,  we will use this as context, otherwise use default context.